teuthology/task/tests/test_fetch_coredumps.py

```diff
@@ -0,0 +1,139 @@
+from teuthology.task.internal import fetch_binaries_for_coredumps
+from unittest.mock import patch
+import gzip
+import os
+
+class TestFetchCoreDumps(object):
+    class MockDecode(object):
+        def __init__(self, ret):
+            self.ret = ret
+            pass
+
+        def decode(self):
+            return self.ret
```

Comment:
Why not use [Mock](https://docs.python.org/3/library/unittest.mock.html#the-mock-class) and [MagicMock](https://docs.python.org/3/library/unittest.mock.html#magicmock-and-magic-method-support)? That way you could, among other things, define separate return values and/or exceptions for each method.

Is there a benefit to all these custom classes?

Reply:
No benefits, I agree that custom classes are not the best way, more like I was exploring different ways to Mock objects.